backend/src/akvo/lumen/auth.clj

```diff
@@ -81,3 +99,41 @@
 (s/def ::keycloak ::keycloak/data)
 (defmethod ig/pre-init-spec :akvo.lumen.auth/wrap-jwt [_]
   (s/keys :req-un [::keycloak]))
+
+(defn api-tenant-admin?
+  [tenant allowed-paths]
+  (contains? allowed-paths (str tenant "/admin")))
+
+(defn api-tenant-member?
+  [tenant allowed-paths]
+  (or
+   (contains? allowed-paths (str tenant "/admin"))
+   (contains? allowed-paths tenant)))
+
+(defmethod ig/init-key :akvo.lumen.auth/wrap-authorization
+  [_ {:keys [keycloak]}]
+  (fn [handler]
+    (fn [{:keys [jwt-claims tenant] :as request}]
+      (if (api-authz? jwt-claims)
+        (try
+          (let [email (get jwt-claims "email")
+                allowed-paths (keycloak/allowed-paths keycloak email)]
```

Comment:
allowed-paths should not run in the  `(nil? jwt-claims)` and  `:else` cases as it is making at least one http request to KeyCloak.

Either wrap it on a `delay` or move it inside the  `(if (api-tenant-admin? tenant allowed-paths)` branches